python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_message_store.py

```diff
@@ -0,0 +1,44 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from ...messages import BaseChatMessage
+
+
+class MessageStore(ABC):
+    """
+    Abstract base class for storing a message thread
+    """
+
+    @abstractmethod
+    async def add_message(self, message: BaseChatMessage) -> None:
+        """
+        Add a message to the store
+        """
+        pass
+
+    @abstractmethod
+    async def add_messages(self, messages: List[BaseChatMessage]) -> None:
```

Comment:
It should be `BaseChatMessage | BaseAgentEvent` to be consistent with how message threads are implemented as list in `BaseGroupChatManager`.